docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,237 @@
+
+- It is easy to setup and configure.
+- We have a Storybook deployment already in place and has worked great so far for a few other FxA services so hooking up the client for `fxa-auth-server` to view the modernized components might result in consistency.
+- Easy to test and perform QA. The design and product teams don't have to touch the codebase to test the various states of UI.
+- Support for CSS (whether it's plain old CSS, TailwindCSS or something else)
+
+### Other
+
+**Plaintext files**: [todo]
+
+**Localization**: We will upgrade the localization tool from GetText to Fluent since it's preferred by the l10n team and other FxA front-ends are using it.
+
+**Integration & QA**: [todo]
+We also intend on making a plan on how we're going to build out/hook refactored templates up with feature flags and mitigate risk. The `fxa-auth-server` already has some basic support for feature flagging, but this ADR aims to identify how we want to use these - do we want 3 different feature flags (one for each milestone of templates), do we want to flag individual templates, or should we have a single flag?
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
+- Cons
+  - The logicless nature of Mustache would continue to bother us in the future
+
+### Option B - Use React server-side to generate static pages
+
+- Description
+  - Template the auth server emails using [React](https://reactjs.org/) server-side and convert them over to static HTML pages. Other pros and cons of React can be found [in this ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0010-transition-fxa-from-backbone-to-react.md#option-c---move-to-isomorphic-react)
+- Pros
+  - The logicless nature of the templates will be eliminated once they're refactored in React
+  - React is capable of rendering pages to [static HTML](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup)
+  - We already use React in `fxa-settings`, `fxa-admin-panel`, `fxa-payments-server`, and `fxa-react`. There’s potential to cross-share components with the new email templates.
+  - React offers us many integration options. E.g there is one with [MJML](https://github.com/wix-incubator/mjml-react)
+  - React has already been audited by our security team.
+  - It may give us an edge over performance since the server will be doing most of the work
```

Comment:
I don't think we'd see a performance boost switching from Mustache to React. 🤔 Mustache also has to be rendered server-side then sent over to the client.